main/design_documents/ARM_NEON_SVE_BRIDGE.rst

```diff
@@ -0,0 +1,86 @@
+=======================================
+Design Document for ARM_NEON_SVE_BRIDGE
+=======================================
+
+This document describes the design of the intrinsics defined in
+`NEON_SVE Bridge <../acle.rst#neon-sve-bridge>`_. These C intrinsics enable the
+conversion between NEON and SVE vectors with no performance overhead, and
+duplicating NEON vectors across SVE vectors.
+
+Location
+########
+
+The intrinsics are defined in ``arm_neon_sve_bridge.h``. If
+``arm_neon_sve_bridge.h`` is available on the compiler,
+``__ARM_NEON_SVE_BRIDGE`` is defined as 1. Users include the NEON/SVE Bridge
+as follows:
+
+.. code:: c
+
+    #ifdef __ARM_NEON_SVE_BRIDGE
+    #include "arm_neon_sve_bridge.h"
+    #endif //__ARM_NEON_SVE_BRIDGE
+
+``arm_neon_sve_bridge.h`` includes ``arm_neon.h`` and ``arm_sve.h``.
+
+Suggestions & Considerations
+############################
+
+Why not enable the extraction of any 128-bit subvector in an SVE vector?
+------------------------------------------------------------------------
+
+Extracting other 128-bit subvectors goes against the no-op design.
+This functionality is already available with the ACLE intrinsic 
+``svdupq_lane``. For example:
+
+``svuint64_t sve_vector_a = ...;``
+
+``uint64x2_t neon_vec = svget_neonq_u64(svdupq_lane_u64(sve_vector_a, 1));``
+
+Why not enable the insertion of a 128-bit NEON vector into a specific SVE subvector?
+------------------------------------------------------------------------------------
+
+arm_sve.h does not currently provide any intrinsics for inserting individual
+elements into a specific lane of an SVE vector. The approach should be
+consistent between ``arm_sve.h`` and ``arm_neon_sve_bridge.h``
+
+Why not include 64-bit Vectors?
+-------------------------------
+
+Support for 64-bit vectors is not added because of low interest. The
+naming scheme is designed to cope with the possibility of supporting these in
+the future. The C standard requires a unique function name for each set of
+input and return types from these functions. C Generics enable inferring the
+intended function from its parameter types, but not from the return type.
+Therefore we adopt the NEON convention of using the letter `q` in the name to
+indicate a 128-bit vector in the return type. e.g. ``svget_neonq[_u16]``. If
+64-bit vectors are added later they will omit the `q`.
+
+Why not shuffle the vector bits for big-endian?
+-----------------------------------------------
+
+Copying vectors bit-for-bit as a Q vector rather than lane-for-lane is safer
+by default for setting, getting and duping. Users requiring lane-shuffling can
+then shuffle the bits as required. Shuffling bridge intrinsics can be added if
+enough interest is shown.
```

Comment:
```suggestion
Copying vectors bit-for-bit as a Q vector rather than lane-for-lane is safer
by default for setting, getting and duplicating lanes. Users requiring lane-shuffling can
then shuffle the bits as required. Shuffling bridge intrinsics can be added if
enough interest is shown.
```